engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/lite/requestresponse/RequestResponseAkkaHttpHandler.scala

```diff
@@ -0,0 +1,98 @@
+package pl.touk.nussknacker.engine.lite.requestresponse
+
+import akka.event.Logging
+import akka.http.scaladsl.model.MediaTypes.`application/json`
+import akka.http.scaladsl.model.{HttpEntity, HttpResponse, ResponseEntity, StatusCodes}
+import akka.http.scaladsl.server.directives.DebuggingDirectives
+import akka.http.scaladsl.server.{Directive0, Directive1, Directives, Route}
+import akka.stream.Materializer
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{NonEmptyList, Validated}
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.generic.JsonCodec
+import io.circe.syntax.EncoderOps
+import io.circe.{Encoder, Json}
+import pl.touk.nussknacker.engine.api.Context
+import pl.touk.nussknacker.engine.api.exception.NuExceptionInfo
+import pl.touk.nussknacker.engine.requestresponse.DefaultResponseEncoder
+import pl.touk.nussknacker.engine.requestresponse.FutureBasedRequestResponseScenarioInterpreter.InterpreterType
+import pl.touk.nussknacker.engine.requestresponse.RequestResponseInterpreter.RequestResponseResultType
+import pl.touk.nussknacker.engine.requestresponse.api.{RequestResponseGetSource, RequestResponsePostSource}
+import pl.touk.nussknacker.engine.requestresponse.metrics.InvocationMetrics
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class ProcessRoute(processInterpreters: scala.collection.Map[String, InterpreterType]) extends Directives with LazyLogging {
+
+  protected def logDirective(scenarioName: String): Directive0 = DebuggingDirectives.logRequestResult((s"request-response-$scenarioName", Logging.DebugLevel))
+
+  def route(implicit ec: ExecutionContext, mat: Materializer): Route =
+    path(Segment) { processPath =>
+      logDirective(processPath) {
+        processInterpreters.get(processPath) match {
```

Comment:
maybe move this part into separate method like process/scenarioInterpreterForPath(process/scenarioPath)

Reply:
:+1: